pkg/controller/build/build_controller.go

```diff
@@ -278,14 +289,26 @@ func (r *ReconcileBuild) validateOutputSecret(ctx context.Context, n string, ns
 		return errors.Errorf("there are no secrets in namespace %s", ns)
 	}
 
-	if len(list.Items) > 0 {
-		for _, secret := range list.Items {
-			if secret.Name == n {
-				return nil
-			}
+	var lookUp = map[string]bool{}
+	for _, secretName := range secretNames {
+		lookUp[secretName] = false
+	}
+	for _, secret := range list.Items {
+		lookUp[secret.Name] = true
+	}
+	var missingSecrets []string
+	for name, found := range lookUp {
+		if !found {
+			missingSecrets = append(missingSecrets, name)
 		}
-		return fmt.Errorf("secret %s does not exist", n)
 	}
+
+	if len(missingSecrets) > 1 {
+		return fmt.Errorf("secrets %s do not exist", strings.Join(missingSecrets, ", "))
+	} else if len(missingSecrets) > 0 {
```

Comment:
I think you can use `len > 0` directly:
```
	if len(missingSecrets) > 0 {
		return fmt.Errorf("secret %s does not exist", strings.Join(missingSecrets, ", "))
	}
```

If it is only one secret in the array, `strings.Join` will show without ", ":
```
	var missingSecrets []string
	missingSecrets = []string{"icr-knbuild"}
	fmt.Printf(strings.Join(missingSecrets, ", "))
```
Result:
```
✔ $ go run main.go 
icr-knbuild
```

Reply:
I know, but then one has a plural `secrets` in the message but only one listed.

Reply:
Not sure if we can use one refined sentence to cover it like:
https://github.com/shipwright-io/build/blob/master/pkg/controller/buildrun/buildrun_controller.go#L483

Reply:
There might be others, but I use [text.Plural](https://github.com/gonvenience/text/blob/main/text.go#L93-L122) to have a helper function to pick the right singular/plural form for me. It is a bit nerdy, I know.

Reply:
But I get the point of not touching the error message, because it is already mentioned somewhere. Since the missing secret is kind of a more likely error case, I could understand the additional code logic to have proper error messages.

Reply:
Yep, it is not a blocker for this PR.

If it looks good now, we can good ahead.